website/docs/externaldata.md

```diff
@@ -0,0 +1,166 @@
+- Addresses common patterns with a single provider, e.g. image tag-to-digest mutation, which can be leveraged by multiple scenarios (e.g. validating image signatures or vulnerabilities).
+- Provider model creates a common interface for extending Gatekeeper with external data.
+  - It allows for separation of concerns between the implementation that allows access to external data and the actual policy being evaluated.
+  - Developers and consumers of data sources can rely on that common protocol to ease authoring of both constraint templates and data sources.
+  - Makes change management easier as users of an external data provider should be able to tell whether upgrading it will break existing constraint templates. (once external data API is stable, our goal is to have that answer always be "no")
+- Performance benefits as Gatekeeper can now directly control caching and which values are significant for caching, which increases the likelihood of cache hits.
+  - For mutation, we can batch requests via lazy evaluation.
+  - For validation, we make batching easier via function design.
+
+## Enabling external data support
+
+### YAML
+You can enable external data support by adding `--enable-external-data` in gatekeeper audit and controller-manager deployment arguments.
+
+### Helm
+You can also enable external data by installing or upgrading Helm chart by setting `enableExternalData=true`:
+
+```sh
+helm install gatekeeper/gatekeeper --name-template=gatekeeper --namespace gatekeeper-system --create-namespace \
+	--set enableExternalData=true \
+	--set controllerManager.dnsPolicy=ClusterFirst,audit.dnsPolicy=ClusterFirst
+```
+
+> Please note that setting dnsPolicy is a workaround for a bug that is already fixed and will be available in a future release.
+
+### Dev/Test
+
+For dev/test deployments, use `make deploy ENABLE_EXTERNAL_DATA=true`
+
+## Providers
+
+Providers are designed to be in-cluster components that can communicate with external data sources (such as image registries, Active Directory/LDAP directories, etc) and return data in a format that can be processed by Gatekeeper.
+
+Example provider can be found at: https://github.com/open-policy-agent/gatekeeper/tree/master/test/externaldata/dummy-provider
+
+## Provider implementation
+
+Provider is an HTTP server that listens on a port and responds to [ProviderRequest](#ProviderRequest) with [ProviderResponse](#ProviderResponse).
+
+As part of [ProviderResponse](#ProviderResponse), the provider can return a list of items. Each item is a JSON object with the following fields:
+- `Key`: the key that was sent to the provider
+- `Value`: the value that was returned from the provider for that key
+- `Error`: an error message if the provider returned an error for that key
+
+If there is a system error, the provider should return the system error message in the `SystemError` field.
+
+> Recommendation is to keep a timeout such as maximum of 1-2 seconds for the provider to respond.
```

Comment:
maybe consider moving this after the provider crd as before the crd, there's no mention of `timeout` field at all.

Reply:
I had it in the comment. added clarification